### DISCUSSION TITLE
Math Expression Renderer inside the Text Editor

### DISCUSSION THREAD

**Discussion by DinosNeverDied** [microsoft/vscode-discussions#1223] [Extension Development QnA] [answered]
I'd like to create an configurable extension for VSCode that renders mathematical expressions in more pleasant ways. This should happen **inside the text editor** - without the need for opening a second file.

Some examples in Python:

| Source Code | Visual Representation |
|-----------------|-----------------|
| a**2          | a²   |
| a_1        | a₁   |
| a.dot(b)          | a · b   |
| (x + 1) / x| $\frac{x+1}{x}$ (in a far future)

I am completey aware of the complexity of this problem and the potential critisism to such invasive changes. However, I believe in the value of experimentation: nothing ventured, nothing gained.

I am wondering: Is such an extension even possible? Are there maybe even multiple ways to archive this? Could this extension potentially win sympathy of users?

**Comment by isidorn** [comment, marked as answer]
Yes. You can achieve this using our Custom Editor API. You can render the visual representation using a WebView
https://code.visualstudio.com/api/extension-guides/custom-editors
https://code.visualstudio.com/api/extension-guides/webview
